server/auto_updater/embedding_bin_updater.py

```diff
@@ -0,0 +1,319 @@
+                / journal_centroid[journal]["counter"]
+            )
+            output_dict = {
+                f"feat_{idx}": mean_vec[idx] for idx in range(mean_vec.shape[0])
+            }
+            output_dict["journal"] = journal
+            writer.writerow(output_dict)
+
+
+def generate_SAUCIE_coordinates(
+    new_papers, paper_landscape_file, paper_landscape_json_file
+):
+    """
+    This function calculates the SAUCIE coordinates for each paper emebdding
+    and then recalculates bins for each newly added paper.
+
+    Parameters:
+        new_papers - the file that contains all the paper embeddings
+        paper_landscape_file - the file that contains papers mapped to bins
+        paper_landscape_json_file - the json file the frontend uses
+    """
+
+    new_papers_df = pd.read_csv(new_papers, sep="\t")
+
+    # Load SAUCIE model and generate coordinates for new papers
+    saucie_model = SAUCIE(300, restore_folder="../saucie_model")
+    coordinates = saucie_model.get_embedding(
+        Loader(new_papers_df[[f"feat_{idx}" for idx in range(300)]].values)
+    )
+
+    # Activate pandas to r dataframe conversion
+    pandas2ri.activate()
+
+    # Grab the SAUCIE updated coordinates
+    landscape_df = pd.read_csv(f"{paper_landscape_file}", sep="\t").append(
+        pd.DataFrame(coordinates, columns=["dim1", "dim2"]).assign(
+            document=new_papers_df.document.tolist(),
+            journal=new_papers_df.journal.tolist(),
+        )[["dim1", "dim2", "journal", "document"]],
+        sort=True,
+    )
+
+    # Bin the data into squares
+    robjects.globalenv["data_df"] = robjects.conversion.py2rpy(landscape_df)
+    robjects.r.source("get_square_bins.R")
+    square_bin_plot_df = robjects.conversion.rpy2py(
+        robjects.globalenv["square_plot_df"]
+    )
+
+    # Iterate through the squares and assign bin_id to data
+    square_iterator = enumerate(square_bin_plot_df.iterrows())
+    mapped_data_df = pd.DataFrame([], columns=landscape_df.columns.tolist())
+
+    for idx, (row_idx, square_bin) in tqdm.tqdm(square_iterator):
+
+        top_left = (square_bin["xmin"], square_bin["ymax"])
+        bottom_right = (square_bin["xmax"], square_bin["ymin"])
+
+        datapoints_df = landscape_df.query(
+            f"dim1 > {top_left[0]} and dim1 < {bottom_right[0]}"
+        ).query(f"dim2 < {top_left[1]} and dim2 > {bottom_right[1]}")
+
+        mapped_data_df = mapped_data_df.append(
+            datapoints_df.assign(squarebin_id=idx).reset_index(drop=True),
+            ignore_index=True,
+            sort=True,
+        )
+
+    # Write updated file to disk
+    (mapped_data_df.to_csv(paper_landscape_file, sep="\t", index=False))
+
+    # Write updated bin plot to disk
+    (
+        square_bin_plot_df.assign(bin_id=mapped_data_df.squarebin_id.unique().shape[0])[
+            ["x", "y", "xmin", "xmax", "ymin", "ymax", "bin_id"]
+        ].to_json("pmc_square_plot.json", orient="records", lines=False)
```

Comment:
I think `"pmc_square_plot.json"` should be replaced by `paper_landscape_json_file`.